Snowflake.Data.Tests/IntegrationTests/StructuredTypesWithEmbeddedUnstructuredIT.cs

```diff
@@ -361,6 +361,41 @@ internal static IEnumerable<object[]> DateTimeConversionCases()
                 null,
                 DateTime.Parse("2024-07-11 21:20:05.1234568").ToLocalTime()
             };
+            yield return new object[]
+            {
+                "9999-12-31 23:59:59.999999",
+                SFDataType.TIMESTAMP_NTZ.ToString(),
+                DateTime.Parse("9999-12-31 23:59:59.999999"),
+                DateTime.Parse("9999-12-31 23:59:59.999999")
+            };
+            yield return new object[]
+            {
+                "9999-12-31 23:59:59.999999 +1:00",
+                SFDataType.TIMESTAMP_TZ.ToString(),
+                null,
+                DateTime.SpecifyKind(DateTime.Parse("9999-12-31 22:59:59.999999"), DateTimeKind.Utc)
+            };
+            yield return new object[]
+            {
+                "9999-12-31 23:59:59.999999 +13:00",
+                SFDataType.TIMESTAMP_LTZ.ToString(),
```

Comment:
maybe we could add timestamp_ltz for early years as well?

Reply:
There is a problem with ToLocalTime() with small dates like `0001-01-01`. It behaves differently on Linux and Windows. On Linux, .NET includes the historical data; on Windows, it does not. That's why I skipped such tests. Check this: https://developercommunity.visualstudio.com/t/Result-from-DateTimeToLocalTime-is-of/10576042